code/obj/item/retribution.dm

```diff
@@ -0,0 +1,233 @@
+						destruction_sds(destruction_point_x, destruction_point_y,destruction_point_z)
+		..()
+
+	proc/destruction_sds(var/point_x, var/point_y, var/point_z)
+		var/create_scan_decal = false
+		var/window_step = 0
+		var/turf/T = locate(point_x,point_y,point_z)
+		for (var/atom/scan_target in T)
+			if (ismob(scan_target))
+				create_scan_decal = true
+				if (isrobot(scan_target))
+					random_burn_damage(scan_target, 15)
+					scan_target.changeStatus("stunned", 2 SECOND)
+				else
+					random_burn_damage(scan_target, 30)
+					scan_target.changeStatus("weakened", 2 SECOND)
+				INVOKE_ASYNC(scan_target, /mob.proc/emote, "scream")
+				playsound(scan_target.loc, "sound/impact_sounds/burn_sizzle.ogg", 70, 1)
+			else if (istype(scan_target, /obj/structure/girder))
+				create_scan_decal = true
+				scan_target.ex_act(1)
+			else if (istype(scan_target, /obj/grille))
+				create_scan_decal = true
+				window_step++
+				scan_target.ex_act(1)
+			else if (istype(scan_target, /obj/window))
+				if(window_step == 0)
+					create_scan_decal = true
+					scan_target.ex_act(1)
+		if(istype(T, /turf/simulated/wall))
+			create_scan_decal = true
+			T = T.ReplaceWith(/turf/simulated/floor/plating/random)
+		if(create_scan_decal)
+			leavescan(T, 1)
+			playsound(T, "sound/effects/smoke_tile_spread.ogg", 50, 1)
+		return
+
+/obj/decal/syndicate_destruction_scan_center
+	name = "Scan"
+	desc = "A glowing hologram, indicating the center of a scan."
+	anchored = 1
+	density = 0
+	opacity = 0
+	icon = null
+	icon_state = null
+	var/image/activation
+
+	New()
+		..()
+		activation = image('icons/misc/retribution/SWORD_loot.dmi', "SDS_tile_activate")
+		activation.plane = PLANE_SELFILLUM
+		src.UpdateOverlays(activation, "activation")
+
+
+/obj/decal/syndicate_destruction_scan_side
+	name = "Scan"
+	desc = "A hardlight hologram, hot to the touch."
+	anchored = 1
+	density = 0
+	opacity = 0
+	icon = null
+	icon_state = null
+	var/image/activation
+
+	New()
+		..()
+		activation = image('icons/misc/retribution/SWORD_loot.dmi', "SDS_tile_scan")
+		activation.plane = PLANE_SELFILLUM
+		src.UpdateOverlays(activation, "activation")
+
+/obj/decal/purge_beam
+	name = "Linear Purge Beam"
+	desc = "A powerful laser. Standing in it's path isn't the wisest of choices."
+	anchored = 1
+	density = 0
+	opacity = 0
+	icon = null
+	icon_state = null
+	var/image/beam
+
+	New()
+		..()
+		beam = image('icons/misc/retribution/SWORD/abilities_o.dmi', "linearPurge_beamBody")
+		beam.plane = PLANE_SELFILLUM
+		src.UpdateOverlays(beam, "beam")
+
+/obj/decal/purge_beam_end
+	name = "Linear Purge Beam"
+	desc = "A powerful laser. Standing in it's path isn't the wisest of choices."
+	anchored = 1
+	density = 0
+	opacity = 0
+	icon = null
+	icon_state = null
+	var/image/beam
+
+	New()
+		..()
+		beam = image('icons/misc/retribution/SWORD/abilities_o.dmi', "linearPurge_beamEnd")
+		beam.plane = PLANE_SELFILLUM
+		src.UpdateOverlays(beam, "beam")
```

Comment:
Any reason why you're doing these decals with overlays rather than icon/icon_state? I've never touched planes before, but couldn't you put the decal itself on the selfillum plane?

Reply:
The last time I tried that, it didn't work. In fact, the fact that overlays work at all is a godsend because for some dumb reason overlays are completely invisible if the icon in question doesn't loop.
Hence why I had to put every single animated icon I made into looping mode.
![image](https://user-images.githubusercontent.com/55758339/107524769-25dc9400-6b84-11eb-90fb-afed5af861bd.png)